### PR TITLE
Handle root delete

### PR DESCRIPTION
**Pull request by aenario**

### CODE REVIEW

File: core/local/watcher.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ class LocalWatcher {
   checksumer: Checksumer
   watcher: any // chokidar
   buffer: LocalEventBuffer<ChokidarFSEvent>
+  ensureInterval: number
```
Comment (sebn) [inline core/local/watcher.js]: `s/ensureInterval/ensureDirInterval`?

File: core/local/watcher.js [hunk 2]
```diff
@@ -67,11 +68,21 @@ class LocalWatcher {
     this.checksumer = checksumer.init()
   }
 
+  ensureDirSync () {
+    // This code is duplicated in local/index#start
+    if (!fs.existsSync(this.syncPath)) {
+      this.events.emit('syncdir-unlinked')
+      throw new Error('Syncdir has been unlinked')
```
Comment (sebn) [inline core/local/watcher.js]: Include `syncPath` in error message?

File: gui/js/onboarding.window.js [hunk 3]
```diff
@@ -36,6 +36,24 @@ module.exports = class OnboardingWM extends WindowManager {
...
+      setTimeout(() => this.send('registration-done')
+      , 20)
+    })
+  }
+
+  create () {
+    return super.create()
+      .then(() => {
+        if (this.shouldJumpToSyncPath) {
+          this.send('registration-done')
```
Comment (sebn) [inline gui/js/onboarding.window.js]: Not sure to understand why...

File: gui/js/onboarding.window.js [hunk 4]
```diff
@@ -36,6 +36,24 @@ module.exports = class OnboardingWM extends WindowManager {
     }
   }
 
+  jumpToSyncPath () {
+    this.shouldJumpToSyncPath = true
+    this.send('registration-done')
+    this.win.once('dom-ready', () => {
+      setTimeout(() => this.send('registration-done')
+      , 20)
```
Comment (sebn) [inline gui/js/onboarding.window.js]: Add `TODO: Ensure elm side sends something through ports so we can trigger 'registration-done' without relying on timeouts`?

File: gui/js/onboarding.window.js [hunk 5]
```diff
@@ -36,6 +36,24 @@ module.exports = class OnboardingWM extends WindowManager {
     }
   }
 
+  jumpToSyncPath () {
+    this.shouldJumpToSyncPath = true
+    this.send('registration-done')
```
Comment (sebn) [inline gui/js/onboarding.window.js]: Remove the first `registration-done` event? Or add a `TODO` notice? Do we have a spinner on the `Use Cozy Drive` onboarding button?

File: gui/js/tray.js [hunk 6]
```diff
@@ -11,6 +11,7 @@ module.exports.init = (app, listener) => {
      : (process.platform === 'win32') ? `${imgs}/tray-icon-win/idle.png`
      : `${imgs}/tray-icon-linux/idle.png`
   tray = new Tray(icon)
+  app.on('before-quit', () => tray.destroy())
```
Comment (sebn) [inline gui/js/tray.js]: May also fix the duplicate tray icon bug on Windows 🎉

File: gui/js/onboarding.window.js (hunk 4 above)
Comment (sebn) [reply gui/js/onboarding.window.js]: (although timeout was already present in code moved from `gui/main`)

File: gui/main.js [hunk 7]
```diff
@@ -85,12 +87,36 @@ const sendErrorToMainWindow = (msg) => {
...
     }
+    trayWindow.hide()
     dialog.showMessageBox(null, options)
     desktop.stopSync()
       .then(() => desktop.removeConfig())
       .then(() => log.info('removed'))
       .then(() => trayWindow.doRestart())
       .catch((err) => log.error(err))
+    return // no notification
+  } else if (msg === 'Syncdir has been unlinked') {
```
Comment (sebn) [inline gui/main.js]: Maybe set some `#code` property on the error so we don't rely on the exact wording?

File: gui/main.js [hunk 8]
```diff
@@ -85,12 +87,36 @@ const sendErrorToMainWindow = (msg) => {
...
+    }
+    trayWindow.hide()
+    dialog.showMessageBox(null, options)
+    desktop.stopSync()
+      .then(() => desktop.pouch.db.destroy())
+      .then(() => { desktop.config.syncPath = undefined })
+      .then(() => desktop.config.persist())
+      .then(() => log.info('removed'))
+      .then(() => trayWindow.doRestart())
+      .catch((err) => log.error(err))
```
Comment (sebn) [inline gui/main.js]: Maybe extract to some `reset()` function?

File: gui/main.js (hunk 7 above)
Comment (aenario) [reply gui/main.js]: in main.js errors are string, so we can't set an error.

File: core/local/watcher.js (hunk 2 above)
Comment (aenario) [reply core/local/watcher.js]: Error can either come in initial start throw or from a later 'syncdir-unlinked' event, which make it hard to handle them both the same.

File: gui/js/onboarding.window.js (hunk 5 above)
Comment (aenario) [reply gui/js/onboarding.window.js]: TODO added

**Comment by aenario** [comment]
I agree there is need for improvement  in main process errors / state management, but I'd prefer to work on it in a separate PR.

**Pull request merged**